server/src/main/java/org/elasticsearch/common/logging/ESLogMessage.java

```diff
@@ -72,4 +81,9 @@ public static String asJsonArray(Stream<String> stream) {
             .map(ESLogMessage::inQuotes)
             .collect(Collectors.joining(", ")) + "]";
     }
+
+    public static ESLogMessage message(String messagePattern, Object... args){
```

Comment:
We have tooling to check that loggers are used correctly which I think is now circumvent-able with this method.  
Most probably the tooling will have to be updated regardless of how this API is implemented.

Reply:
you probably mean `ESLoggerUsageTests` ? I think it should be fine as it was already extended for subclass usages. But I will have a look to make sure it passes fine

Reply:
Yes I do mean the `ESLoggerUsageChecker` I don't know enough about it's implementation to be able to tell if it will work with the builders or not.

Reply:
@pgomulka Could we maybe add an explicit test to `ESLoggerUsageChecker` that uses `ESLogMessage.of`  just to be clear that this works ?